tests/pulp_test.py

```diff
@@ -0,0 +1,23 @@
+import testinfra
+
+def test_pulp_service(host):
+    pulp = host.service("pulp")
+    assert pulp.is_running
+    assert pulp.is_enabled
+
+def test_pulp_port(host):
+    pulp = host.addr("localhost")
+    assert pulp.port("8080").is_reachable
+
+def test_pulp_process_running(host):
+    pulp_process = host.process.filter(comm="pulp")
+    assert pulp_process.exists
+
+# def test_pulp_container(host):
+#     container_info = host.run("podman inspect pulp").stdout
+#     assert "pulp/pulp:stable" in container_info
+#     assert "Running" in container_info
+
+def test_systemd_reload(host):
+    result = host.run("systemctl daemon-reload")
+    assert result.succeeded
```

Comment:
This tests you can run `daemon-reload`, but that doesn't assert any behavior we want. What you want to assert is that a deamon-reload isn't actually needed (for example, because it already ran). https://www.baeldung.com/linux/systemctl-daemon-reload tells me `systemctl show pulp --property=NeedDaemonReload` should return `NeedDaemonReload=no`, but I didn't test this with containers. You also want to try out what it returns when the `pulp.container` doesn't exist, though that's probably already covered by your `test_pulp_service` test. And https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.service.Service tells me you can even extend it with `assert pulp.systemd_properties['NeedDaemonReload'] == 'no'`, which makes it really clean.